drivers/led/led_gpio.c

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <drivers/gpio.h>
+#include <drivers/led.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(led_gpio, CONFIG_LED_GPIO_LOG_LEVEL);
+
+#define DT_DRV_COMPAT gpio_leds
+#define LED_DATA(node_id)                      \
+{                                              \
+	{                                      \
+		DT_GPIO_LABEL(node_id, gpios), \
+		DT_GPIO_PIN(node_id, gpios),   \
+		DT_GPIO_FLAGS(node_id, gpios)  \
+	},                                     \
+	DT_LABEL(node_id),                     \
+	DT_PROP(node_id, id)                   \
+},
+#define LED_SW_BLINK BIT(0)
+#define LED_BLINK_DISABLE BIT(1)
+
+struct gpio_cfg {
```

Comment:
The `pin` field here should be type `gpio_pin_t`.  The `polarity` flag here should be named `flags` (or `dt_flags`) and be of type `gpio_dt_flags_t`.

I don't see why this is a separate structure rather than part of `led_dts_cfg`.

Reply:
I changed `pin`->`gpio_pin_t` and `polarity` -> `gpio_dt_flags_t`. I used a separate structure for readability